### ISSUE TITLE
EmrEtlRunner: Spark Enrich: Failed to fetch 2.0.0 jar

### ISSUE THREAD

**Issue by zehua** [snowplow/snowplow#4286]
**Project**: EmrEtlRunner / Spark Enrich

**Version**: r118 / 2.0.0

**Expected behavior**: The runner should be able to download the spark enrich jar file and start the enrichment job.

**Actual behavior**: The runner failed to download the jar file from s3 and the entire job failed. Exception:

```
2020-02-24T21:15:47.676Z INFO Ensure step 3 jar file s3://snowplow-hosted-assets-us-east-1/3-enrich/scala-hadoop-enrich/snowplow-hadoop-enrich-2.0.0.jar
INFO Failed to download: s3://snowplow-hosted-assets-us-east-1/3-enrich/scala-hadoop-enrich/snowplow-hadoop-enrich-2.0.0.jar
java.lang.RuntimeException: Error whilst fetching 's3://snowplow-hosted-assets-us-east-1/3-enrich/scala-hadoop-enrich/snowplow-hadoop-enrich-2.0.0.jar'
	at aws157.instancecontroller.util.S3Wrapper.fetchS3HadoopFileToLocal(S3Wrapper.java:412)
	at aws157.instancecontroller.util.S3Wrapper.fetchHadoopFileToLocal(S3Wrapper.java:351)
	at aws157.instancecontroller.master.steprunner.HadoopJarStepRunner$Runner.<init>(HadoopJarStepRunner.java:243)
	at aws157.instancecontroller.master.steprunner.HadoopJarStepRunner.createRunner(HadoopJarStepRunner.java:152)
	at aws157.instancecontroller.master.steprunner.HadoopJarStepRunner.createRunner(HadoopJarStepRunner.java:146)
	at aws157.instancecontroller.master.steprunner.StepExecutor.runStep(StepExecutor.java:136)
	at aws157.instancecontroller.master.steprunner.StepExecutor.run(StepExecutor.java:70)
	at aws157.instancecontroller.master.steprunner.StepExecutionManager.enqueueStep(StepExecutionManager.java:246)
	at aws157.instancecontroller.master.steprunner.StepExecutionManager.doRun(StepExecutionManager.java:193)
	at aws157.instancecontroller.master.steprunner.StepExecutionManager.access$000(StepExecutionManager.java:33)
	at aws157.instancecontroller.master.steprunner.StepExecutionManager$1.run(StepExecutionManager.java:94)
Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: The specified key does not exist. (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey; Request ID: 84641F353605B937), S3 Extended Request ID: 8Y70lnITLhCHM82DzldNyj25yyq5V9UriFvZZGNNXMv5GkPznZuNcmTRD0sDwOIjhdVtWhUDSKw=
	at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1389)
	at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:902)
	at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:607)
	at com.amazonaws.http.AmazonHttpClient.doExecute(AmazonHttpClient.java:376)
	at com.amazonaws.http.AmazonHttpClient.executeWithTimer(AmazonHttpClient.java:338)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:287)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:3826)
	at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1143)
	at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1021)
	at aws157.instancecontroller.util.S3Wrapper.copyS3ObjectToFile(S3Wrapper.java:303)
	at aws157.instancecontroller.util.S3Wrapper.getFile(S3Wrapper.java:287)
	at aws157.instancecontroller.util.S3Wrapper.fetchS3HadoopFileToLocal(S3Wrapper.java:399)
	... 10 more
```

**Steps to reproduce**:

 1. Fill the [sample yml](https://github.com/snowplow/snowplow/blob/3b8d9cc839e4af0b97c68477fb1c9f484de233e2/3-enrich/emr-etl-runner/config/config.yml.sample) file with usual values
 2. Update `spark_enrich` to `2.0.0` (https://github.com/snowplow/snowplow/blob/3b8d9cc839e4af0b97c68477fb1c9f484de233e2/3-enrich/emr-etl-runner/config/config.yml.sample#L68)
 3. Run the job with r118

**Cause**

The check in `utils.rb` in line 139 does not handle `2.0.0`. 
https://github.com/snowplow/snowplow/blob/44a27697b22ca196d43a3fa5bb87e90538cbe44d/3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/utils.rb#L139 

It could be changed to:

```
version[0] > 1 || version[0] == 1 && version[1] >= 9
```

**Comment by chuwy** [comment]
Hi @zehua,

Sorry for the confusion, it seems we forgot to remove Spark Enrich commits and [items](https://github.com/snowplow/snowplow/blob/master/CHANGELOG#L3-L6) from CHANGELOG before mergine a release branch in master, but we did not intend to release Spark Enrich with R118. In fact we're actively [deprecating all batch pipeline components](https://discourse.snowplowanalytics.com/t/rfc-making-the-snowplow-pipeline-real-time-end-to-end-and-deprecating-support-for-batch-processing-modules/3018), including Spark Enrich and recommend everyone to [migrate to real-time pipeline](https://discourse.snowplowanalytics.com/t/aws-batch-pipeline-to-real-time-pipeline-upgrade-guide/3470).

In other words, there will be likely no Spark Enrich 2.0.0, the final version was never published.
